lib/routes.js

```diff
@@ -1,212 +1,248 @@
+const select_routes_by_site = query.bind(query, fs.readFileSync('./sql/select_routes_by_site.sql').toString('utf8'), to_response);
+const select_route_by_details = query.bind(query, fs.readFileSync('./sql/select_route_by_details.sql').toString('utf8'), to_response);
+const select_routes_by_pending = query.bind(query, fs.readFileSync('./sql/select_routes_by_pending.sql').toString('utf8'), to_response);
+const update_route = query.bind(query, fs.readFileSync('./sql/update_route.sql').toString('utf8'), to_response);
+const delete_route_query = query.bind(query, fs.readFileSync('./sql/delete_route.sql').toString('utf8'), to_response);
+const insert_route = query.bind(query, fs.readFileSync('./sql/insert_route.sql').toString('utf8'), to_response);
 
 function check_route(route) {
-  assert.ok(/(^[A-z0-9_\/\.-]+$)/.test(route.source_path), 'The source_path of a route must match [A-z0-9_/-\.]+.');
-  if(route.target_path !== '') {
-    assert.ok(/(^[A-z0-9_\/\.-]+$)/.test(route.target_path), 'The target_path of a route must match [A-z0-9_/-]+.');
+  assert.ok(/(^[A-z0-9_/.-]+$)/.test(route.source_path), 'The source_path of a route must match [A-z0-9_/-.]+.');
+  if (route.target_path !== '') {
+    assert.ok(/(^[A-z0-9_/.-]+$)/.test(route.target_path), 'The target_path of a route must match [A-z0-9_/-]+.');
   }
-  assert.ok(route.site, 'The route must attach to a site. Please specify site ID or domain name.')
+  assert.ok(route.site, 'The route must attach to a site. Please specify site ID or domain name.');
 }
 
 async function push_routes(pg_pool, region, domain) {
-  return await common.alamo.sites.push_routes(pg_pool, region, domain)
+  return common.alamo.sites.push_routes(pg_pool, region, domain);
 }
 
 async function remove_alamo_path(pg_pool, region, domain, source_path, donotpush) {
-  let data = await common.alamo.sites.delete_route(pg_pool, region, domain, source_path)
-  if(!donotpush) {
-    await push_routes(pg_pool, region, domain)
+  const data = await common.alamo.sites.delete_route(pg_pool, region, domain, source_path);
+  if (!donotpush) {
+    await push_routes(pg_pool, region, domain);
   }
-  return data
+  return data;
 }
 
 async function delete_route(pg_pool, route, donotpush) {
-  let site = await common.site_exists(pg_pool, route.site.id)
-  let routes = await delete_route_query(pg_pool, [route.id])
+  const site = await common.site_exists(pg_pool, route.site.id);
+  const routes = await delete_route_query(pg_pool, [route.id]);
   if (routes.length === 0) {
-    throw new common.NotFoundError(`The specified route does not exist.`);
+    throw new common.NotFoundError('The specified route does not exist.');
   }
-  if(route.pending !== true) {
-    return await remove_alamo_path(pg_pool, site.region_name, site.domain, route.source_path, donotpush)
+  if (route.pending !== true) {
+    return remove_alamo_path(pg_pool, site.region_name, site.domain, route.source_path, donotpush);
   }
+  return undefined;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unexpected use of undefined.](https://app.codacy.com/manual/Akkeris/controller-api/pullRequest?prid=5244943)